packages/components/src/pagination-control/index.tsx

```diff
@@ -0,0 +1,43 @@
+import { __, sprintf } from '@wordpress/i18n';
+import classnames from 'classnames';
+import { times } from 'lodash';
+import './style.scss';
+
+interface Props {
+	currentPage: number;
+	numberOfPages: number;
+	setCurrentPage: ( page: number ) => void;
+	classNames?: string | string[];
+}
+
+const PaginationControl: React.FunctionComponent< Props > = ( props ) => {
+	const { currentPage, numberOfPages, setCurrentPage, classNames, children } = props;
+
+	const classes = classnames( 'pagination-controls', classNames?.toString()?.split( ',' ) );
+
+	return (
+		<ul className={ classes } aria-label={ __( 'Guide controls' ) }>
+			{ times( numberOfPages, ( page ) => (
+				<li key={ page } aria-current={ page === currentPage ? 'page' : undefined }>
+					<button
+						key={ page }
+						className={ classnames( 'pagination-control__page', {
+							'pagination-control__current': page === currentPage,
+						} ) }
+						disabled={ page === currentPage }
+						aria-label={ sprintf(
+							/* translators: 1: current page number 2: total number of pages */
+							__( 'Page %1$d of %2$d' ),
+							page + 1,
+							numberOfPages
+						) }
+						onClick={ () => setCurrentPage( page ) }
+					/>
+				</li>
+			) ) }
+			{ children && <li className={ classnames( 'navigation-buttons' ) }>{ children }</li> }
```

Comment:
We don't need the call to `classnames` here, just add it as a string. The linter will/should also match it to coding standards (my guess is we'll rename it to `pagination-control__navigation-buttons`).

Reply:
Also, does this need to be part of the same `ul` list? It might be better structure-wise to sit in its own separate element and just wrap this along with the list? 🤔

Reply:
>Also, does this need to be part of the same ul list? It might be better structure-wise to sit in its own separate element and just wrap this along with the list? 🤔

It could, but if the user of the component wanted that then couldn't they just use the component without children and build the `div` to their liking? I think having children go into the `ul` allows for adding other elements into the list which could be helpful not just buttons. Maybe the class should be more ambiguous to reflect that?

Reply:
Ah that makes sense :-)

>  Maybe the class should be more ambiguous to reflect that?

Yeap, good idea then!